File: apps/server/internal/application/dto/presence.go
package dto

import "time"

// SendPresenceRequest represents a request to send presence update
type SendPresenceRequest struct {
	SessionID string `json:"session_id" validate:"required"`
	ChatJID   string `json:"chat_jid" validate:"required"`
	State     string `json:"state" validate:"required,oneof=typing paused online offline"`
}

// PresenceResponse represents the response after sending presence
type PresenceResponse struct {
	ChatJID   string    `json:"chat_jid"`
	State     string    `json:"state"`
	Timestamp time.Time `json:"timestamp"`
}

File: packages/schema/src/presence.ts
/**
 * Presence-related schemas
 * Matches backend: dto/presence.go
 */
import { z } from "zod";
import {
  jidSchema,
  sessionIdSchema,
  presenceStateSchema,
  timestampSchema,
} from "./common";

/**
 * Send presence request schema
 * Matches: dto.SendPresenceRequest
 */
export const sendPresenceRequestSchema = z.object({
  session_id: sessionIdSchema,
  chat_jid: jidSchema,
  state: presenceStateSchema,
});

export type SendPresenceRequest = z.infer<typeof sendPresenceRequestSchema>;

/**
 * Presence response schema
 * Matches: dto.PresenceResponse
 */
export const presenceResponseSchema = z.object({
  chat_jid: jidSchema,
  state: presenceStateSchema,
  timestamp: timestampSchema,
});

export type PresenceResponse = z.infer<typeof presenceResponseSchema>;
